includes/admin/meta-boxes/class-wc-meta-box-product-data.php

```diff
@@ -446,6 +448,35 @@ public static function save_variations( $post_id, $post ) {
 			$max_loop   = max( array_keys( wp_unslash( $_POST['variable_post_id'] ) ) ); // phpcs:ignore WordPress.Security.ValidatedSanitizedInput.InputNotSanitized
 			$data_store = $parent->get_data_store();
 			$data_store->sort_all_product_variations( $parent->get_id() );
+			$new_variation_menu_order_id = ! empty( $_POST['new_variation_menu_order_id'] ) ? wc_clean( wp_unslash( $_POST['new_variation_menu_order_id'] ) ) : false;
+			$new_variation_menu_order_value = ! empty( $_POST['new_variation_menu_order_value'] ) ? wc_clean( wp_unslash( $_POST['new_variation_menu_order_value'] ) ) : false;
+
+			// Only perform this operation if setting menu order via the prompt.
+			if ( $new_variation_menu_order_id && $new_variation_menu_order_value ) {
+				/*
+				 * We need to gather all the variations with menu order that is
+				 * equal or greater than the menu order that is newly set and
+				 * increment them by one so that we can correctly insert the updated
+				 * variation menu order.
+				 */
+				$ids = $wpdb->get_col(
+					$wpdb->prepare(
+						"SELECT ID FROM {$wpdb->posts} WHERE post_type = 'product_variation' AND post_parent = %d AND post_status = 'publish' AND menu_order >= %d AND ID != %d ORDER BY menu_order ASC",
+						$post_id,
+						$new_variation_menu_order_value,
+						$new_variation_menu_order_id
+					)
+				);
+
+				foreach ( $ids as $id ) {
+					$wpdb->query(
+						$wpdb->prepare(
+							"UPDATE {$wpdb->posts} SET menu_order = menu_order + 1 WHERE ID = %d",
+							$id
+						)
+					);
+				}
```

Comment:
Is there any reason to not do this as a single query?

Something like this?...

```suggestion
				$wpdb->prepare(
					"UPDATE {$wpdb->posts} SET menu_order = menu_order + 1 WHERE post_type = 'product_variation' AND post_parent = %d AND post_status = 'publish' AND menu_order >= %d AND ID != %d",
					$post_id,
					$new_variation_menu_order_value,
					$new_variation_menu_order_id
				);
```

Reply:
Good find. So in my original solution the order mattered `ORDER BY menu_order ASC` but not anymore in this case so I think a single query is a good idea. Thanks.